internal/services/domain/resource_domain_role_assignments.go

```diff
@@ -335,10 +332,30 @@ func (r *resourceDomainRoleAssignments) Delete(ctx context.Context, req resource
 	}
 
 	_, err := r.client.RoleAssignmentsBulkUnassign(ctx, state.DomainID.ValueString(), reqDelete.DomainRoleUnassignmentRequest, nil)
-	if resp.Diagnostics.Append(utils.GetDiagsFromError(ctx, err, utils.OperationDelete, nil)...); resp.Diagnostics.HasError() {
+	diags = utils.GetDiagsFromError(ctx, err, utils.OperationDelete, fabcore.ErrDomain.DomainSpecificUsersScopeCannotBeEmptyError)
```

Comment:
I get why we're doing this here, but it feels a bit like deciding the flow of execution based on the type of errors we receive (like doing try-catch with specific type of exception in some other languages.) Are you sure we cannot infer it would happen, based on the current assignments and scope and the updated values, instead of trying and failing first?

Reply:
It may be the case that a few roles are managed outside of Terraform, and during the deletion operation on our end, Terraform removes only managed roles and leaves the rest - this is expected behavior.
`DomainSpecificUsersScopeCannotBeEmptyError` only occurs when all managed roles on the Terraform side equal roles on the services side. When this happens, the API returns this error, indicating that it is safe to use this workaround without failing.